src/nnet3/nnet-chain-combine.cc

```diff
@@ -1,6 +1,7 @@
 // nnet3/nnet-chain-combine.cc
 
-// Copyright 2012-2015   Johns Hopkins University (author: Daniel Povey)
+// Copyright 2012-2017   Johns Hopkins University (author: Daniel Povey)
```

Comment:
I don't think it's necessary to retain nnet-chain-combine.{h,cc} or nnet-combine.{h,cc} if the combination stage is being simplified like this, because this structure (having a class to do it) is overkill.  You could just put the code directly in the binaries, or at most, have a function declared in nnet-utils.h.
But you do need to compute the objf because you need to select how many models to average over.
Look at the the code in nnet3-average.cc to see how to average over n models.  To compute a running average you don't have to re-do the sum every time and I think you don't even need to keep all the models in memory.  Just read them one by one, you will only have to keep 2 or 3 models in memory I think which will avoid problems with GPU memory.   
And bear in mind that, IIRC, the python scripts reverse the order in which the C++ binary sees the models.

But if, to prototype it, it's eaiser to modify this code, that's OK.